### PR TITLE
Fix fusion tab issue 6741

### PR DESCRIPTION
**Pull request by khzhu**
Fix issue (https://github.com/cBioPortal/cbioportal/issues/6741)
Describe changes proposed in this pull request:
- resolving conflicts when merging hyve fusion implementation to the cBioPortal codebase
- not displaying the locations if no exome/transcripts present

# Checks
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). It can help to figure out who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them either through slack or by assigning them as a reviewer on the PR

### CODE REVIEW

File: src/pages/resultsView/fusion/Fusions.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2018 The Hyve B.V.
```
Comment (onursumer) [inline src/pages/resultsView/fusion/Fusions.tsx]: Are this file and similar files with the same header, included by this PR or is it just a rebase issue?

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 2]
```diff
@@ -2721,6 +2727,70 @@ export class ResultsViewPageStore {
...
+            return client.fetchStructuralVariantsUsingPOST({
+                'structuralVariantFilter': getParamsForStructuralVariants(this.genes.result, this.selectedMolecularProfiles.result, this.samples.result)
+            })
+        }
+    });
+
+    readonly fusionsByGene = remoteData<{[hugeGeneSymbol: string]: StructuralVariantExt[]}>({
+        await: () => [this.genes, this.fusions],
+        invoke: () => {
+            const fusionsByGene = _(this.genes.result!)
```
Comment (onursumer) [inline src/pages/resultsView/ResultsViewPageStore.ts]: might be better to move longer logic into some utility function

**Comment by alisman** [comment]
There is no fusion data in structural variant table in public db to test this out.  So, @n1zea144 is helping us to load some SV data so we can test this PR against it.

File: env/custom.sh [hunk 3]
```diff
@@ -1,2 +1,2 @@
 # set e.g.
-# export CBIOPORTAL_URL="http://localhost:8080"
+export CBIOPORTAL_URL="http://localhost:8082/cbioportal"
```
[the remark this answers is not in the document]
Comment (khzhu) [reply env/custom.sh]: sorry accidentally pushed back.Will revoke.

File: src/pages/resultsView/ResultsViewPageStore.ts (hunk 2 above)
Comment (khzhu) [reply src/pages/resultsView/ResultsViewPageStore.ts]: good point! Will change.

**Comment by inodb** [comment]
@khzhu this was a temporary fix right? But since we're close to releasing 3.2.0, we can maybe close this PR or should it still be merged to release 3.2.0? Thanks!

**Comment by khzhu** [comment]
@inodb , the PR is a fixing for the issue we have when re-importing a study with fusion/structural variant data. It supports a hybrid mode so old style fusion data can co-exist with new structural variant data. Please get it merged with release 3.2.0. Thank you!

**Comment by khzhu** [comment]
@inodb , I thought the plan was not including Fusion tab (structural variants visualization) in release 3.2.0. It is not a temporary fixing and does the following:
1. merging hyve structural variant implementation to the code base of cbioportal
2. fixing fusion table crash when no break points supplied.

**Comment by inodb** [comment]
Thanks @khzhu , i see so it supersedes: https://github.com/cBioPortal/cbioportal-frontend/pull/1361

**Comment by inodb** [comment]
closing this as superseded by #3023

**Pull request closed without merging**